dataset/src/test/scala/frameless/CreateTests.scala

```diff
@@ -28,6 +28,66 @@ class CreateTests extends TypedDatasetSuite with Matchers {
       Vector[(Food, Country)]] _))
   }
 
+  test("array fields") {
+
+    def prop[T](implicit arb: Arbitrary[Array[T]], encoder: TypedEncoder[X1[Array[T]]]) = forAll {
+      data: Array[T] =>
+        val Seq(X1(arr)) = TypedDataset.create(Seq(X1(data))).collect().run()
+        Prop(arr.sameElements(data))
+    }
+
+    check(prop[Boolean])
+    check(prop[Byte])
+    check(prop[Short])
+    check(prop[Int])
+    check(prop[Long])
+    check(prop[Float])
+    check(prop[Double])
+    check(prop[X1[String]])
```

Comment:
Would be nice to have combinations of `X1`, `Option`, `Int` and `String` as well.